packages/editor/src/components/provider/index.js

```diff
@@ -1,34 +1,58 @@
 /**
  * External dependencies
  */
-import { flow, map } from 'lodash';
+import { map } from 'lodash';
 
 /**
  * WordPress Dependencies
  */
 import { compose } from '@wordpress/compose';
-import { createElement, Component } from '@wordpress/element';
-import { DropZoneProvider, SlotFillProvider } from '@wordpress/components';
-import { withDispatch } from '@wordpress/data';
+import { Component } from '@wordpress/element';
+import { withDispatch, withSelect } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
+import { BlockEditorProvider } from '@wordpress/block-editor';
 
 /**
  * Internal dependencies
  */
 import { traverse, wrap, urlRewrite } from '../../editor-styles';
 
+/**
+ * Compute the Editor Provider's state object from props.
+ * @param {Object} props Component props.
+ *
+ * @return {Object} Component state.
+ */
+function computeProviderStateFromProps( props ) {
+	return {
+		settings: props.settings,
```

Comment:
On react blog it says:
> If you’re using derived state to memoize some computation based only on the current props, you don’t need derived state. See What about memoization? below.

I think in this case we are only using the derived state to make sure the editorSettings we pass references the same object if the props it depends on did not change.
I think we can use memoization and just memoize a getEditorSettings function, similar to the sample available in https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#what-about-memoization.

Reply:
You're right. It seems like they're using `memoize-one`, should we use `memize` with a size of 1 to mimic this? cc @aduth